src/utils/attribution-html.ts

```diff
@@ -1,70 +1,191 @@
-  return `<p style="font-size: 0.9rem;font-style: italic;">${mediaTag}${imgLink}${creator}${licenseLink}${licenseImgLink}</p>`
+/**
+ * This interface describes the options that the `getAttribution` function can
+ * take to customise its output.
+ */
+interface AttributionOptions {
+  includeIcons?: boolean
+  isPlaintext?: boolean
 }
 
+/* Actual util */
+
 /**
- * Get the HTML markup for attributing the given media item.
+ * Get the HTML markup for properly attributing the given media item.
  *
- * @param mediaItem - the media item, image or audio from the API, to attribute
+ * @param mediaItem - the media item being attributed
+ * @param i18n - the i18n instance to access translations
+ * @param includePreview - whether to include the preview markup in the HTML
  * @returns the HTML markup of the attribution
  */
-export const getAttributionMarkup = (mediaItem: Media) => {
+export const getAttribution = (
+  mediaItem: AttributableMedia,
+  i18n: VueI18n | null = null,
+  { includeIcons, isPlaintext }: AttributionOptions = {
+    isPlaintext: false,
+    includeIcons: true,
+  }
+): string => {
   if (!mediaItem) return ''
 
-  const extLink = (href: string, text: string) =>
-    `<a rel="noopener noreferrer" target="_blank" href="${href}">${text}</a>`
+  /* Title */
 
-  let title = `"${mediaItem.title}"`
-  if (mediaItem.foreign_landing_url) {
-    title = extLink(mediaItem.foreign_landing_url, title)
+  let titleLink = mediaItem.title || ''
+  if (!isPlaintext && mediaItem.foreign_landing_url)
+    titleLink = extLink(mediaItem.foreign_landing_url, titleLink)
+
+  /* Creator */
+
+  let creatorLink = mediaItem.creator || ''
+  if (!isPlaintext && mediaItem.creator_url)
+    creatorLink = extLink(mediaItem.creator_url, creatorLink)
+
+  /* License */
+
+  const fullLicenseName = getFullLicenseName(
+    mediaItem.license,
+    mediaItem.license_version,
+    i18n
+  )
+  let licenseIcons = ''
+  if (includeIcons && mediaItem.license) {
+    const elements = getElements(mediaItem.license)
+    const icons = elements.map((element) => licenseElementImg(element))
+    // Icons are only rendered if present for every element
+    if (!icons.includes('')) {
+      licenseIcons = icons.join('')
+    }
   }
 
-  let creator = mediaItem.creator
-  if (mediaItem.creator_url) {
-    creator = extLink(mediaItem.creator_url, creator)
+  let licenseLink = `${fullLicenseName} ${licenseIcons}`.trim()
+  if (!isPlaintext && mediaItem.license_url) {
+    licenseLink = extLink(`${mediaItem.license_url}?ref=openverse`, licenseLink)
   }
 
-  let license = `${mediaItem.license} ${mediaItem.license_version}`
-  license = extLink(mediaItem.license_url, license.toUpperCase())
+  /* Attribution */
 
-  return `${title} by ${creator} is licensed under ${license}.`
-}
+  const i18nBase = 'media-details.reuse.credit'
+  const isPd = isPublicDomain(mediaItem.license)
+
+  let attribution: string
+  if (i18n) {
+    let fillers: object = {
```

Comment:
Why the `object` type annotation? Unfortunately this type is mostly non-useful for 99% of applications.

Did you want `Record<string, any>` maybe?

Reply:
Yes, updated in d7b9bde.